django/db/models/deletion.py

```diff
@@ -51,6 +52,23 @@ def DO_NOTHING(collector, field, sub_objs, using):
     pass
 
 
+def get_candidate_relations_to_delete(opts):
+    # Collect models that contain candidate relations to delete. This may include
+    # relations coming from proxy models.
+    candidate_models = {opts}
+    candidate_models = candidate_models.union(opts.concrete_model._meta.proxied_children)
+    # For each model, get all candidate fields.
+    candidate_model_fields = chain.from_iterable(
+        opts.get_fields(include_hidden=True) for opts in candidate_models
+    )
+    # Keep related objects not coming from ManyToManyFields from all candidate
+    # model fields.
+    return (
+        f for f in candidate_model_fields
+        if f.auto_created and not f.concrete and not f.many_to_many
```

Comment:
Firstly - my apologies, I meant ManyToOneField, not OneToManyField as my candidate example.

As for the proposed change: I'm of two minds. Any discussion around this topic involves a certain amount of architecture astronauting, because we're discussing the needs of a completely theoretical class of fields that doesn't exist in the Real World (AFAICT). 

@collinanderson's approach is explicit on the fields that we currently have, but will obviously be weaker *if* other field types are defined. @PirosB3's approach is stronger from a theoretical perspective, because it tries to capture the concept, rather than the specific, but we don't have anything to verify whether the concept is actually correct in practice, other than a ManyToOneField that doesn't exist. As @collinanderson points out, the ManyToOneField demonstrates that the currently logic isn't correct as written; I'm sure there's a more correct way of expressing the abstract idea, but it isn't immediately obvious to me.

So - we have three options AFAICT:
 * Keep it as is, because It Works(tm), and *might* be right for a certain class of extension fields.
 * Wring our hands trying to fix the logic for an theoretical case that doesn't exist.
 * Change to @collinanderson's suggestion, on the basis that it's more explicit, and treat any problems identified with extension fields as bugs to be addressed over time.

Thoughts?